From a6b4b9b4bcf2c970b35d439a68e1332da1869c53 Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Thu, 20 Oct 2016 10:57:22 +0800
Subject: [PATCH 136/188] net: macb: Enable SG mode support for zynq

This commit 4c16a042 comes from:
  https://github.com/Xilinx/linux-xlnx.git

Changes: add #ifdef to scope this only to Zynq series.

Since zynq ethernet controller supports scatter gather IO, Enabling
this feature for zynq.

Signed-off-by: Punnaiah Choudary Kalluri <punnaia@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/cadence/macb.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/net/ethernet/cadence/macb.c b/drivers/net/ethernet/cadence/macb.c
index d954a97..bd4ad60 100644
--- a/drivers/net/ethernet/cadence/macb.c
+++ b/drivers/net/ethernet/cadence/macb.c
@@ -2804,7 +2804,11 @@ static const struct macb_config at91sam9260_config = {
 };
 
 static const struct macb_config pc302gem_config = {
+#if defined(CONFIG_ARCH_ZYNQ)
+	.caps = MACB_CAPS_GIGABIT_MODE_AVAILABLE,
+#else
 	.caps = MACB_CAPS_SG_DISABLED | MACB_CAPS_GIGABIT_MODE_AVAILABLE,
+#endif
 	.dma_burst_length = 16,
 	.clk_init = macb_clk_init,
 	.init = macb_init,
-- 
2.0.2

